src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -104,7 +114,7 @@
 import static org.apache.cassandra.utils.Throwables.merge;
 import static org.apache.cassandra.utils.Throwables.perform;
 
-public class ColumnFamilyStore implements ColumnFamilyStoreMBean, Memtable.Owner
+public class ColumnFamilyStore implements ColumnFamilyStoreMBean, Memtable.Owner, CompactionRealm
```

Comment:
If that fails, it should not be hard to make a wrapper which has a similar mode of operation to the `TableDataSupplier`, but I think it would be easier to directly implement the realm using the view/tracker.

Reply:
I think using the view/tracker directly should be doable. Resolving for now.